operator/pkg/gateway-api/cell.go

```diff
@@ -221,58 +226,131 @@ func checkCRD(ctx context.Context, clientset k8sClient.Clientset, gvk schema.Gro
 	return nil
 }
 
-func checkRequiredCRDs(ctx context.Context, clientset k8sClient.Clientset) error {
+// checkCRDs checks if required and optional CRDs are present in the cluster,
+// returns an error if the required CRDs are not installed, and returns the
+// schema.GroupVersionKind of any optional CRDs that are installed.
+func checkCRDs(ctx context.Context, clientset k8sClient.Clientset, logger *slog.Logger) ([]schema.GroupVersionKind, error) {
 	var res error
-	for _, gvk := range requiredGVK {
+	var presentGVKs []schema.GroupVersionKind
+
+	for _, gvk := range requiredGVKs {
 		if err := checkCRD(ctx, clientset, gvk); err != nil {
 			res = errors.Join(res, err)
 		}
 	}
-	return res
+
+	for _, optionalGVK := range optionalGVKs {
+		if strings.HasSuffix(optionalGVK.Kind, "lists") {
```

Comment:
do we really need to check this exception? IMO it would be just wrong to add the list-type to the list of optional GVKs.

Reply:
Yes, this was leftover from a previous version of `checkCRD()` that did not do the list-adding itself.